xarray_beam/_src/pangeo_test.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for xarray_beam._src.pangeo."""
+
+import contextlib
+import itertools
+import tempfile
+
+import numpy as np
+from pangeo_forge_recipes.patterns import FilePattern, ConcatDim
+
+from xarray_beam._src import core
+from xarray_beam._src import test_util
+from xarray_beam._src.pangeo import FilePatternToChunks, _zero_dimensions, \
+  _expand_dimensions_by_key
+
+
+class ExpandDimensionsByKeyTest(test_util.TestCase):
+  TEST_DATA = test_util.dummy_era5_surface_dataset()
+  LEVEL = ConcatDim("level", list(range(91, 100)))
+  PATTERN = FilePattern(lambda level: f"gs://dir/{level}.nc", LEVEL)
```

Comment:
Please put these variables in a `setUp()` or `setUpClass()` method instead of directly defining class variables.

The setup helper methods are nice because they allow for importing the test file without immediately doing non-trivial execution. This makes things like test discovery work better, e.g., in case something goes wrong in setup.

Reply:
Sure thing, and I like that reasoning.